### PR TITLE
kinto init command

### PR DESCRIPTION
**Pull request by oak11**

### CODE REVIEW

**Comment by oak11** [comment]
The kinto init command code runs template.py to generate a .ini file in config folder

File: config/kinto.ini [hunk 1]
```diff
@@ -11,20 +11,20 @@ use = egg:kinto
...
 # http://kinto.readthedocs.org/en/latest/configuration/settings.html#storage
-#
-# kinto.storage_backend = cliquet.storage.postgresql
-# kinto.storage_url = postgres://postgres:postgres@localhost/postgres
-# kinto.cache_backend = cliquet.cache.postgresql
-# kinto.cache_url = postgres://postgres:postgres@localhost/postgres
-# kinto.permission_backend = cliquet.permission.postgresql
-# kinto.permission_url = postgres://postgres:postgres@localhost/postgres
+[backends]
+kinto.storage_backend = cliquet.storage.postgresql
```
Comment (Natim) [inline config/kinto.ini]: please remove trailling spaces.

Comment (Natim) [inline config/kinto.ini]: Also maybe we should not commit the generated file anymore.

File: config/template.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import os
+import binascii
+
+def render_template(template, destination, **kwargs):
+
+
+   here = os.path.abspath(os.path.dirname(__file__))
```
Comment (Natim) [inline config/template.py]: nit pick: This could be a constant (HERE in uppercase) defined at the beginning of the file.

Comment (Natim) [inline config/template.py]: I think the template.py file should not be in the config folder but in the kinto/ one.

File: config/template.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+   template = os.path.join(here, 'kinto.tpl')
```
Comment (Natim) [inline config/template.py]: Please change to:

    template = os.path.join(here, '..', 'config', 'kinto.tpl')

File: config/template.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import os
+import binascii
+
+def render_template(template, destination, **kwargs):
+
+
+   here = os.path.abspath(os.path.dirname(__file__))
+   template = os.path.join(here, 'kinto.tpl') 
+   destination = os.path.join(here, 'kinto.ini')
```
Comment (Natim) [inline config/template.py]: Same here:

    destination = os.path.join(here, '..', 'config', 'kinto.ini')

File: config/template.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+    with open(template, 'r') as f:
```
Comment (Natim) [inline config/template.py]: We should probably use ``codecs.open`` to handle UTF-8 in this files.

See: https://github.com/mozilla-services/cliquet/blob/master/setup.py#L8-L9

File: config/template.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+       
+        rendered = raw_template.format(**kwargs)
+
+        with open(destination, 'w+') as output:
+            output.write(rendered)
+
+
+def init():
+	values= {}
+	values['secret'] = binascii.b2a_hex(os.urandom(32))
```
Comment (Natim) [inline config/template.py]: :+1:

File: config/template.py [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+	backend = raw_input("Which backend to use ? (1 - in-memory, 2 - postgresql, 3 - redis) :")
```
Comment (Natim) [inline config/template.py]: Please use ``six.moves.input`` to make this code compatible with both Python2 and Python3.

http://pythonhosted.org/six/#module-six.moves

File: config/template.py [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+	if backend == '':
+	    backend = '2';     #default
```
Comment (Natim) [inline config/template.py]: I would rather change line 29 into:

    if backend in ('2', ''):

File: config/template.py [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+
+	if backend == '2':
```
Comment (Natim) [inline config/template.py]: What about redis and memory setup?

File: config/template.py [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+	    
+	    render_template("kinto.tpl", "kinto.ini",
+		            secret = values['secret'],
+		            storage_backend = values['storage_backend'],
+		            storage_url = values['storage_url'],
+		            cache_backend = values['cache_backend'],
+		            cache_url = values['cache_url'],
+		            permission_backend = values['permission_backend'],
+		            permission_url = values['permission_url']
+		            )
```
Comment (Natim) [inline config/template.py]: Indentation here is not right, we should put the parenthesis at the end of the previous line

File: kinto/__main__.py [hunk 11]
```diff
@@ -3,6 +3,7 @@
 from cliquet.scripts import cliquet
 from pyramid.scripts import pserve
 from pyramid.paster import bootstrap
+from config import template
```
Comment (Natim) [inline kinto/__main__.py]: please import it like ``from kinto.config import template``

File: MANIFEST.in [hunk 12]
```diff
@@ -1,2 +1,2 @@
 include *.rst LICENSE Makefile tox.ini .coveragerc
-recursive-include config *.ini
+recursive-include config *.ini *.tpl
```
Comment (Natim) [inline MANIFEST.in]: Please include ``kinto/config/kinto.tpl``

**Comment by Natim** [comment]
Why do we have two PR for the same thing ( https://github.com/Kinto/kinto/pull/267 )

Can we close one of them?

**Comment by oak11** [comment]
what values should the variables be assigned for memory setup for backend? Also, I was unable to remove the kinto.ini file from the config folder as the execution halts at __main__.py and gives an error: kinto.ini not found. I have made the other changes.

**Comment by Natim** [comment]
Superseeded by https://github.com/Kinto/kinto/pull/278

**Pull request closed without merging**